plugin/src/main/java/org/opensearch/ml/utils/RestActionUtils.java

```diff
@@ -226,13 +233,26 @@ public static boolean isSuperAdminUser(ClusterService clusterService, Client cli
                 logger.debug("{} is registered as an admin dn", dn);
                 adminDn.add(new LdapName(dn));
             } catch (final InvalidNameException e) {
-                logger.error("Unable to parse admin dn {}", dn, e);
+                logger.debug("Unable to parse admin dn {}", dn, e);
+                adminUsernames.add(dn);
             }
         }
 
-        ThreadContext threadContext = client.threadPool().getThreadContext();
-        final String sslPrincipal = threadContext.getTransient(OPENDISTRO_SECURITY_SSL_PRINCIPAL);
-        return isAdminDN(sslPrincipal);
+        Object userObject = client.threadPool().getThreadContext().getTransient(OPENDISTRO_SECURITY_USER);
+        if (userObject == null)
+            return false;
+        ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
Rather than creating an ObjectMapper for every execution can this be part of class variable?

Reply:
makes sense